test/unit/annotation_layer_spec.js

```diff
@@ -581,6 +581,159 @@ describe('Annotation layer', function() {
     });
   });
 
+  describe('ChoiceWidgetAnnotation', function() {
+    var choiceWidgetDict;
+
+    beforeEach(function (done) {
+      choiceWidgetDict = new Dict();
+      choiceWidgetDict.set('Type', Name.get('Annot'));
+      choiceWidgetDict.set('Subtype', Name.get('Widget'));
+      choiceWidgetDict.set('FT', Name.get('Ch'));
+
+      done();
+    });
+
+    afterEach(function () {
+      choiceWidgetDict = null;
+    });
+
+    it('should handle missing option arrays', function() {
+      var choiceWidgetRef = new Ref(122, 0);
+      var xref = new XRefMock([
+        { ref: choiceWidgetRef, data: choiceWidgetDict, }
+      ]);
+
+      var choiceWidgetAnnotation = annotationFactory.create(xref,
+                                                            choiceWidgetRef);
```

Comment:
Here, and in the other tests, you may also want to explicitly check that you actually get the expected annotation type, e.g. like this:
```javascript
expect(data.annotationType).toEqual(AnnotationType.WIDGET);
```